app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,110 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework import status
+from rest_framework.test import APIClient
+from urllib.parse import urlencode
+
+from recipe.models import Recipe, Ingredient
+
+from recipe.serializers import RecipeSerializer
+
+
+def get_recipies_url(recipe_id=None):
+    if recipe_id:
+        return reverse('recipe:recipe-detail', args=[recipe_id])
+    return reverse('recipe:recipe-list')
+
+
+def url_with_querystring(**kwargs):
+    return get_recipies_url() + '?' + urlencode(kwargs)
+
+
+def create_recipe(**params):
+    defaults = {
+        "name": "Sample recipe name",
+        "description": "Sample description",
+    }
+    defaults.update(params)
+
+    recipe = Recipe.objects.create(**defaults)
+    return recipe
+
+
+class RecipeAPITests(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_retrive_recipes(self):
+        create_recipe()
+        create_recipe()
+
+        res = self.client.get(get_recipies_url())
+
+        recipes = Recipe.objects.all().order_by("-id")
+        serializer = RecipeSerializer(recipes, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
```

Comment:
I think it would be a bit better to compare data with values, not `serializer.data` It's a bit hard to tell what's exactly is being tested here.